src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalysisScope.cs

```diff
@@ -85,6 +86,21 @@ private AnalysisScope(IEnumerable<SyntaxTree> trees, IEnumerable<AdditionalText>
             Debug.Assert(isPartialAnalysis || FilterSpanOpt == null);
             Debug.Assert(isPartialAnalysis || !isSyntacticSingleFileAnalysis);
 
+            if (filterSpanOpt.HasValue)
+            {
+                Debug.Assert(filterFile.HasValue);
+                Debug.Assert(filterFile.Value.SourceTree != null);
+
+                // PERF: Clear out filter span if the span length is equal to the entire tree span, and the filter span starts at 0.
+                //       We are basically analyzing the entire tree, and clearing out the filter span
+                //       avoids span intersection checks for each symbol/node/operation in the tree
+                //       to determine if it falls in the analysis scope.
+                if (filterSpanOpt.Value.Start == 0 && filterSpanOpt.Value.Length == filterFile.Value.SourceTree.Length)
```

Comment:
>Value [](http://example.com/codeflow?start=34&length=5)

Consider using ```GetValueOrDefault``` instead in order to bypass a null check that we already performed..